src/EditorFeatures/Core/Shared/Utilities/ClassificationTypeMap.cs

```diff
@@ -23,19 +26,23 @@ internal class ClassificationTypeMap
             // Prepopulate the identity map with the constant string values from ClassificationTypeNames
             var fields = typeof(ClassificationTypeNames).GetFields();
             _identityMap = new Dictionary<string, IClassificationType>(fields.Length, ReferenceEqualityComparer.Instance);
+
             foreach (var field in fields)
             {
-                var value = (string)field.GetValue(null);
-
                 // The strings returned from reflection do not have reference-identity
                 // with the string constants used by the compiler. Fortunately, a call
                 // to string.Intern fixes them.
-                value = string.Intern(value);
+                var value = string.Intern((string)field.GetValue(null));
                 _identityMap.Add(value, registryService.GetClassificationType(value));
             }
         }
 
         public IClassificationType GetClassificationType(string name)
+        {
+            return GetClassificationTypeWorker(name) ?? GetClassificationTypeWorker(ClassificationTypeNames.Text);
```

Comment:
:bulb: It looks like my previous comments were unclear. I was thinking that only one of the two calls to `ReportWithoutCrash` would get removed, and the other would still cover all the cases. Since we are not expecting a failure in this code, it seemed to make sense to consolidate the error conditions to a single location that reports and recovers from the possible error(s).

Here's what I had in mind for this method:

```csharp
public IClassificationType GetClassificationType(string name)
{
  var classificationType = GetClassificationTypeWorker(name);
  if (classificationType == null)
  {
    FatalError.ReportWithoutCrash(new Exception($"classification type doesn't exist for {value}"));
    classificationType = GetClassificationTypeWorker(ClassificationTypeNames.Text);
  }

  return classificationType;
}
```

If there is concern that `GetClassificationTypeWorker(ClassificationTypeNames.Text)` could return null, the type could be updated to also import `IStandardClassificationService` and use one of its properties as the fallback.

Reply:
sure I can put the NFW back there.